src/kokkos/plugin-PixelTriplets/kokkos/CAHitNtupletGeneratorKernels.cc

```diff
@@ -34,116 +34,166 @@ namespace KOKKOS_NAMESPACE {
-        device_hitTuple_apc_,
-        device_hitToTuple_apc_,  // needed only to be reset, ready for next kernel
-        hh.view(),
-        device_theCells_.get(),
-        device_nCells_,
-        device_theCellNeighbors_,
-        device_isOuterHitOfCell_.get(),
-        m_params.hardCurvCut_,
-        m_params.ptmin_,
-        m_params.CAThetaCutBarrel_,
-        m_params.CAThetaCutForward_,
-        m_params.dcaCutInnerTriplet_,
-        m_params.dcaCutOuterTriplet_);
-    cudaCheck(cudaGetLastError());
+    int nthTot = 64;
+    int stride = 4;
+    int teamSize = nthTot / stride;
+    int leagueSize = (3 * m_params.maxNumberOfDoublets_ / 4 + teamSize - 1) / teamSize;
+    int rescale = leagueSize / 65536;
+    teamSize *= (rescale + 1);
+    leagueSize = (3 * m_params.maxNumberOfDoublets_ / 4 + teamSize - 1) / teamSize;
+    assert(leagueSize < 65536);
+    assert(teamSize > 0 && 0 == teamSize % 16);
+    teamSize *= stride;
+
+    // Current multi-team algorithms are usually based on CUDA implementations which
+    // may not be flexible enough to become a generic parallel solution for all backends.
+    // Thus team policy should be manually handled for each specific backend.
+#ifdef KOKKOS_BACKEND_CUDA
+    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, leagueSize, teamSize};
+#else  // serial
+    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, 1, Kokkos::AUTO()};
+#endif
+    const auto *hhp = hh.view();
+    Kokkos::parallel_for(
+        "kernel_connect", policy, KOKKOS_LAMBDA(const Kokkos::TeamPolicy<KokkosExecSpace>::member_type &teamMember) {
+          kernel_connect(device_hitTuple_apc_,
+                         device_hitToTuple_apc_,  // needed only to be reset, ready for next kernel
+                         hhp,
+                         device_theCells_,
+                         device_nCells_,
+                         device_theCellNeighbors_,  // not used at the moment
+                         device_isOuterHitOfCell_,
+                         m_params.hardCurvCut_,
+                         m_params.ptmin_,
+                         m_params.CAThetaCutBarrel_,
+                         m_params.CAThetaCutForward_,
+                         m_params.dcaCutInnerTriplet_,
+                         m_params.dcaCutOuterTriplet_,
+                         stride,
+                         teamMember);
+        });
 
     if (nhits > 1 && m_params.earlyFishbone_) {
-      auto nthTot = 128;
-      auto stride = 16;
-      auto blockSize = nthTot / stride;
-      auto numberOfBlocks = (nhits + blockSize - 1) / blockSize;
-      dim3 blks(1, numberOfBlocks, 1);
-      dim3 thrs(stride, blockSize, 1);
-      fishbone<<<blks, thrs, 0, cudaStream>>>(
-          hh.view(), device_theCells_.get(), device_nCells_, device_isOuterHitOfCell_.get(), nhits, false);
-      cudaCheck(cudaGetLastError());
+#ifdef KOKKOS_BACKEND_CUDA
+      int teamSize = 128;
+      int stride = 16;
+      int blockSize = nthTot / stride;
```

Comment:
Should this be 
```suggestion
      int blockSize = teamSize / stride;
```
?

Reply:
You are right, sorry about this stupid bug.